### PR TITLE
TM docs for last of stress update models

### PR DESCRIPTION
**Pull request by sapitts**
Adds documentation for the last undocumented radial return stress update material model in tensor mechanics and the last two undocumented TM post processors most likely to be used in Bison/Grizzly simulations.

The regolded temperature dependent regression test reflects the change in the aux variable name from `total_strain_yy` to `strain_yy` that occurred when I updated the test to use the tensor mechanics master action.

@gambka would you please review this content before I ask @aeslaughter or @jwpeterson to merge this pull request?

Refs #10516

### CODE REVIEW

File: modules/tensor_mechanics/doc/content/documentation/systems/Materials/TemperatureDependentHardeningStressUpdate.md [hunk 1]
```diff
@@ -1,19 +1,81 @@
...
+indices as defined for [eqn:hardening_function].
+The $T_{hf}$ value is used in [eqn:hardening_function] to interpolate the
+hardening values.
+
+This class, `TemperatureDependentHardeningStressUpdate`, calculates an effective
+trial stress, an effective scalar plastic strain increment, and the derivative
+of the scalar effective plastic strain increment; these values are passed to the
+[RadialReturnStressUpdate](/RadialReturnStressUpdate.md) to compute the radial
+return stress increment.  This temperature-dependent hardening isotropic
+plasticity class also computes the plastic strain as a stateful material property.
```
Comment (gambka) [inline modules/tensor_mechanics/doc/content/documentation/systems/Materials/TemperatureDependentHardeningStressUpdate.md]: The previous sentence is defines the class I do not think you need to repeat 

> This temperature-dependent hardening isotropic
> plasticity class also computes the plastic strain as a stateful material property

I the sentence would be just fine as `The plastic strain is computed as a stateful material property`

File: modules/tensor_mechanics/include/materials/TemperatureDependentHardeningStressUpdate.h [hunk 2]
```diff
@@ -38,18 +38,35 @@ class TemperatureDependentHardeningStressUpdate : public IsotropicPlasticityStre
...
+  /**
+   * Determines the section of the piecewise temperature dependent hardening
+   * function for the current temperature and calculates the relative temperature
+   * fraction within that piecewise function section.
+   */
   void initializeHardeningFunctions();
 
   MooseSharedPointer<LinearInterpolation> _interp_yield_stress;
+
+  ///@{The function names and expressions for hardening as a functio of temperature
```
Comment (gambka) [inline modules/tensor_mechanics/include/materials/TemperatureDependentHardeningStressUpdate.h]: functio -> function

File: modules/tensor_mechanics/include/materials/TemperatureDependentHardeningStressUpdate.h [hunk 3]
```diff
@@ -38,18 +38,35 @@ class TemperatureDependentHardeningStressUpdate : public IsotropicPlasticityStre
...
   std::vector<Real> _hf_temperatures;
+
+  ///@{Indices to identify the lower and upper temperature bounds for the current value
   unsigned int _hf_index_lo;
   unsigned int _hf_index_hi;
+  ///@}
+
+  /**
+   * The fraction of the temperature within the bounds of the relevant section
+   * of the piecewise hardening function. This hardening temperature faction is
```
Comment (gambka) [inline modules/tensor_mechanics/include/materials/TemperatureDependentHardeningStressUpdate.h]: faction -> fraction

**Review by gambka — CHANGES_REQUESTED** [review]

**Comment by sapitts** [comment]
Comments addressed; thanks for the review @gambka

**Comment by sapitts** [comment]
@aeslaughter or @jwpeterson  would one of you please review this pull request for merging?

**Review by jwpeterson — APPROVED** [review]

**Comment by sapitts** [comment]
Thanks @jwpeterson

**Pull request merged**